tests/channels.rs

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+
+use ethtool::{
+    EthtoolAttr, EthtoolChannelAttr, EthtoolCmd, EthtoolHeader, EthtoolMessage,
+};
+use netlink_packet_generic::{GenlBuffer, GenlHeader};
+use netlink_packet_utils::{Emitable, Parseable, ParseableParametrized};
+
+#[test]
+fn test_channels_get_reply() {
+    let raw: Vec<u8> = vec![
+        0x12, 0x01, 0x00, 0x00, 0x18, 0x00, 0x01, 0x80, 0x08, 0x00, 0x01, 0x00,
+        0x02, 0x00, 0x00, 0x00, 0x09, 0x00, 0x02, 0x00, 0x65, 0x74, 0x68, 0x30,
+        0x00, 0x00, 0x00, 0x00, 0x08, 0x00, 0x05, 0x00, 0x04, 0x00, 0x00, 0x00,
+        0x08, 0x00, 0x09, 0x00, 0x02, 0x00, 0x00, 0x00,
+    ];
+
+    let expected = EthtoolMessage {
+        cmd: EthtoolCmd::ChannelGetReply,
+        nlas: vec![
+            EthtoolAttr::Channel(EthtoolChannelAttr::Header(vec![
+                EthtoolHeader::DevIndex(2),
+                EthtoolHeader::DevName("eth0".to_string()),
+            ])),
+            EthtoolAttr::Channel(EthtoolChannelAttr::CombinedMax(4)),
+            EthtoolAttr::Channel(EthtoolChannelAttr::CombinedCount(2)),
+        ],
+    };
+
+    assert_eq!(
+        expected,
+        EthtoolMessage::parse_with_param(
+            &raw[4..],
+            GenlHeader::parse(&GenlBuffer::new(&raw)).unwrap()
+        )
+        .unwrap(),
+    );
```

Comment:
Can you add `assert_eq!(expected.emit(), raw)`?
This covers the emit trait implementation.

Reply:
I initially wasn't sure whether the header should also be emitted. I changed the test to compare the complete `raw` to the complete expected emitted bytes including header.

Reply:
@cathay4t Is there anything else I can add or change?

Reply:
Looks good to me now.